tests/integration/test_sharding.py

```diff
@@ -148,24 +150,58 @@ def validate_slots(cluster_slots):
         local_id = "{}00000001".format(leader.info()['raft_dbid']).encode()
 
         for i in range(len(cluster_slots)):
-            shargroup_id = cluster_slots[i][2][2]
+            node_id = cluster_slots[i][2][2]
             start_slot = cluster_slots[i][0]
             end_slot = cluster_slots[i][1]
 
-            if shargroup_id == local_id:
+            if node_id == local_id:
                 assert start_slot == 0, cluster_slots
                 assert end_slot == 5, cluster_slots
-            elif shargroup_id == b"1234567890123456789012345678901234567890":
+            elif node_id == b'2' * 40:
                 assert start_slot == 6, cluster_slots
                 assert end_slot == 7, cluster_slots
-            elif shargroup_id == b"1234567890123456789012345678901334567890":
+            elif node_id == b'3' * 40:
                 assert start_slot == 8, cluster_slots
                 assert end_slot == 16383, cluster_slots
             else:
-                assert False, "failed to match id {}".format(shargroup_id)
+                assert False, "failed to match id {}".format(node_id)
 
     validate_slots(cluster.node(3).client.execute_command('CLUSTER', 'SLOTS'))
 
+    def validate_nodes(cluster_nodes):
```

Comment:
based on this test / validation, might want to define a shardgroup with a single slot and no slots.  i.e. validate that the no slot doesn't exist.

Reply:
see single slot validation below